helpers/playerHelper.js

```diff
@@ -2,21 +2,31 @@ var _ = require('lodash');
 var teamHelper = require('./teamHelper.js');
 
 exports.mapPlayer = function(player) {
-  var playerObj = {};
-  var key;
-
-  _.forEach(player, function(obj) {
-    key = _.keys(obj)[0];
-    if ( !_.isUndefined(key) ) {
-      playerObj[key] = obj[key];
-    }
-  });
+  var mergeObjects = function(arrayOfObjects) {
+    var destinationObj = {};
+    var key;
+
+    _.forEach(arrayOfObjects, function(obj) {
+      _.forEach(_.keys(obj), function(key) {
+        if (!_.isUndefined(key)) {
+          destinationObj[key] = obj[key];
+        }
+      });
+    });
+
+    return destinationObj;
+  };
+
+  var playerObj = mergeObjects(player);
 
   playerObj.eligible_positions = _.map(
     playerObj.eligible_positions,
     function(p) { return p.position; }
   );
 
+  playerObj.selected_position = mergeObjects(playerObj.selected_position);
+  playerObj.starting_status = mergeObjects(playerObj.starting_status);
+
```

Comment:
Good call. Maybe not. Really just gotta know how Yahoo! is representing their xml models as json, because that seems to be where it gets screwy. Maybe they blow out all objects, and any object we parse as json needs to be flattenned? No clue.

Regardless, could just check for null in the mergeObjects() function in case they aren't there? Or force the client to do the parsing? I saw you try to simplify the eligible_positions field, so I thought it was a good idea.